### PR TITLE
Create ResistanceSystem.md

### PR DESCRIPTION
**Pull request by rjdurlin42**

### CODE REVIEW

File: Documentation/Signoffs/ResistanceSystem.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+Team 4 – Mario Kart & Ride Replay Simulator
+Team Members: Ray Durlin, Blake Pickett, Tyler Chittum, Benjamin Reed, and Sage Mooneyham
+Detail Design: Resistance System
+
+
+Function of the Subsystem:
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ResistanceSystem.md]: Use markdown headings by starting the line with a pound sign. Add additional pound signs to reduce the size of the header.

File: Documentation/Signoffs/ResistanceSystem.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
+Team 4 – Mario Kart & Ride Replay Simulator
+Team Members: Ray Durlin, Blake Pickett, Tyler Chittum, Benjamin Reed, and Sage Mooneyham
+Detail Design: Resistance System
+
+
+Function of the Subsystem: 
+The function of the resistance system is to wirelessly receive an input value from the RRS processing subsystem. The input value is to be received by the H-Bridge controller. The H-Bridge controller will send a power signal to the linear actuator in the form of a pulse width modulated value to adjust the position of the magnets across the flywheel resulting in the creation of eddy currents in the flywheel. These eddy currents create a magnetic field that opposes the source magnetic field. This causes power dissipation in the flywheel, as well as opposing torque, which creates a change of resistance felt by the user.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ResistanceSystem.md]: The idea is to state what it is supposed to do at a functional level. This is good but it is a bit more low level than is needed here.

File: Documentation/Signoffs/ResistanceSystem.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+Detail Design: Resistance System
+
+
+Function of the Subsystem: 
+The function of the resistance system is to wirelessly receive an input value from the RRS processing subsystem. The input value is to be received by the H-Bridge controller. The H-Bridge controller will send a power signal to the linear actuator in the form of a pulse width modulated value to adjust the position of the magnets across the flywheel resulting in the creation of eddy currents in the flywheel. These eddy currents create a magnetic field that opposes the source magnetic field. This causes power dissipation in the flywheel, as well as opposing torque, which creates a change of resistance felt by the user. 
+
+Constraints:
+Cost: The main constraint for the resistance system is the cost of materials for both the flywheel, linear motor, and magnets. The total budgeted cost of the resistance system is 50% of the project budget, equaling $600. 
+-	Flywheel: A traditional spin bike flywheel used commercially ranges from $270 to $475 used. To mitigate the cost of the flywheel, team 4 will construct a flywheel using the existing bike tire and aluminum sheet material.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ResistanceSystem.md]: This seems like a bad idea. Have you consulted any mechanical engineering staff?

File: Documentation/Signoffs/ResistanceSystem.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
+Team 4 – Mario Kart & Ride Replay Simulator
+Team Members: Ray Durlin, Blake Pickett, Tyler Chittum, Benjamin Reed, and Sage Mooneyham
+Detail Design: Resistance System
+
+
+Function of the Subsystem: 
+The function of the resistance system is to wirelessly receive an input value from the RRS processing subsystem. The input value is to be received by the H-Bridge controller. The H-Bridge controller will send a power signal to the linear actuator in the form of a pulse width modulated value to adjust the position of the magnets across the flywheel resulting in the creation of eddy currents in the flywheel. These eddy currents create a magnetic field that opposes the source magnetic field. This causes power dissipation in the flywheel, as well as opposing torque, which creates a change of resistance felt by the user. 
+
+Constraints:
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ResistanceSystem.md]: This constraint list is not appropriate. It does not consider any of the things that the resistance system actually does (provide resistance, change resistance, etc).

File: Documentation/Signoffs/ResistanceSystem.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+The flywheel thickness analysis took all possible thickness sizes from the Cut2Size custom cut functionality versus the quantity of magnets selected (Cut2Size.com citation). The values of interest, being those within 15% of the torque threshold, were highlighted in red and taken to price analysis.
+The final analysis was the price analysis. This was executed to determine which of the three combinations of flywheel thickness and magnet quantity was the least expensive. It was determined that the least expensive option was the 1.6 mm thick flywheel with four magnets on each side.
+
+
+![image](https://user-images.githubusercontent.com/114370750/201012188-e285c507-e496-41f7-a01a-832c86b08c23.png)
+
+
+
+BOM:
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ResistanceSystem.md]: Needs to be markdown. Not an image.

File: Documentation/Signoffs/ResistanceSystem.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+Buildable schematic:
+
+![image](https://user-images.githubusercontent.com/114370750/201011044-d22a8e48-4f29-43f6-9733-91bda5719795.png)
+
+![image](https://user-images.githubusercontent.com/114370750/201011097-6f954d7b-ff05-4396-bd05-49222ed99d69.png)
+
+
+
+
+Analysis:
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ResistanceSystem.md]: Until the constraints are updated I can't provide feedback on the analysis. The analysis should be an analysis of the design showing that it should be able to meet the constraints. But since the constraints don't address significant core functionality, this must be addressed before feedback is given on analysis.

File: Documentation/Signoffs/ResistanceSystem.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
+Team 4 – Mario Kart & Ride Replay Simulator
+Team Members: Ray Durlin, Blake Pickett, Tyler Chittum, Benjamin Reed, and Sage Mooneyham
+Detail Design: Resistance System
+
+
+Function of the Subsystem: 
+The function of the resistance system is to wirelessly receive an input value from the RRS processing subsystem. The input value is to be received by the H-Bridge controller. The H-Bridge controller will send a power signal to the linear actuator in the form of a pulse width modulated value to adjust the position of the magnets across the flywheel resulting in the creation of eddy currents in the flywheel. These eddy currents create a magnetic field that opposes the source magnetic field. This causes power dissipation in the flywheel, as well as opposing torque, which creates a change of resistance felt by the user. 
+
+Constraints:
+Cost: The main constraint for the resistance system is the cost of materials for both the flywheel, linear motor, and magnets. The total budgeted cost of the resistance system is 50% of the project budget, equaling $600.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ResistanceSystem.md]: This is not a constraint. A constraint is a specific line that must not be crossed. What is your price constraint?

File: Documentation/Signoffs/ResistanceSystem.md (hunk 4 above)
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ResistanceSystem.md]: The discussion of constraints is not relevant or complete. It is impossible to analyze a system that does not have clear expectations regarding the constraints and the required function.

**Review by JesseTNRoberts — COMMENTED** [review]
Constraints are missing. Insufficient analysis. Further, analysis isn't possible without constraints and requirements clearly stated. 

The analysis is a measure of how close you have gotten to a target. The constraints and requirements are the target. How can you measure without knowing the target?

**Comment by JesseTNRoberts** [comment]
I checked this and did not see any changes.

File: Documentation/Signoffs/ResistanceSystem.md [hunk 8]
```diff
@@ -0,0 +1,203 @@
+# Team 4 – Mario Kart & Ride Replay Simulator
+
+## Team Members: Ray Durlin, Blake Pickett, Tyler Chittum, Benjamin Reed, and Sage Mooneyham ##
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ResistanceSystem.md]: Maybe reduce this from a header to just text

File: Documentation/Signoffs/ResistanceSystem.md [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
...ctuator to control the direction of the actuator's motion, and the on-board microcontroller controls the duty cycle of the applied voltage to modulate speed. The microcontroller takes an analog input from the linear actuator to determine its current position, as well as an external digital input (via the Bluetooth module) to determine the calculated set point which the actuator should try to reach. The linear motor is attached to an array of N52 neodymium magnets. The linear motor is to vary the array of magnets perpendicular to the simulator’s rear conductive flywheel. As the magnet array is varied perpendicular to the conductive flywheel, eddy currents are generated by the primary magnetic induction formed at the conductor [1]. As the eddy currents form, braking torque is also generated due to the opposing magnetic field. This torque produced in the flywheel is then transferred to the rear bike wheel. The braking torque functions as the resistance felt by the user. The braking torque ranges from 0 to 28.9 N∙m based on the input resistance level received by the H-Bridge controller. 28.9 N∙m is the greatest amount of torque available given the user is pedaling at the minimum speed.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ResistanceSystem.md]: This is too much. State the intended function briefly and accurately. Bullet points are great.

File: Documentation/Signoffs/ResistanceSystem.md [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+
+**_Constraints:_**
+
+_Flywheel Speed:_
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ResistanceSystem.md]: I think I understand what you are saying but it is not clear.

File: Documentation/Signoffs/ResistanceSystem.md [hunk 11]
```diff
@@ -0,0 +1,203 @@
...
+
+•	The minimum flywheel speed required for the resistance system to operate as designed is 9 mph (or 2501.6 flywheel RPM). At flywheel speeds less than 2501.6 RPM, the resistance system will be unable to reach the maximum torque specification, but it will be controlled to provide resistance in as much as possible given the decreasing braking torque with decreasing angular velocity. The value of 9 mph is based on the American Council on Exercise's recommendation regarding the minimum speed for moderate exercising [2].
+
+•	The maximum speed at which the resistance system will operate is pedal 118 rpm (or 6190.3 Flywheel RPM). The maximum speed is determined based on a cyclist traveling at an elite level [3]. When the flywheel is spinning faster than 6190.3 RPM, the resistance system will minimize braking torque to the flywheel to avoid any injury to the rider.
+
+_Flywheel Weight:_
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ResistanceSystem.md]: why? arbitrary?

File: Documentation/Signoffs/ResistanceSystem.md [hunk 12]
```diff
@@ -0,0 +1,203 @@
...
+
+The flywheel construct is required to weigh 20 pounds. The specified flywheel weight is based on data provided by the bike academy that states many indoor cyclists avoid exercise bikes with flywheels over 20 pounds due to the heavier flywheel making it more difficult to generate a desirable momentum [4].
+
+_Resistance States:_
+
+Based on a study conducted by Romagnoli and Piacentini [5], it was found that test subjects could accurately perceive differences in exercise velocity, and in turn resistance changes, approximately 80% of the time, meaning they were inaccurate 20% of the time. However, there is no published documentation that has determined the exact human threshold a human can sense changes in resistance. Using half of the 20% value, a MATLAB script was created to determine what angles resulted in 10% ± 5% increases in resistance. This resulted in 85 unique states. Because of this, the resistance system shall have no less than 85 states of resistance. 
+
+_Maximum Produced Torque:_
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ResistanceSystem.md]: Probably not a relevant citation. What direction was that force applied? A force constraint based on the force to break a leg is not really relevant as a first line of thought. This should be primarily based on the resistance needed to accomplish the objective. Then, you can add mitigations to avoid these concerns.

File: Documentation/Signoffs/ResistanceSystem.md [hunk 13]
```diff
@@ -0,0 +1,203 @@
...
+
+_Flywheel Weight:_
+
+The flywheel construct is required to weigh 20 pounds. The specified flywheel weight is based on data provided by the bike academy that states many indoor cyclists avoid exercise bikes with flywheels over 20 pounds due to the heavier flywheel making it more difficult to generate a desirable momentum [4].
+
+_Resistance States:_
+
+Based on a study conducted by Romagnoli and Piacentini [5], it was found that test subjects could accurately perceive differences in exercise velocity, and in turn resistance changes, approximately 80% of the time, meaning they were inaccurate 20% of the time. However, there is no published documentation that has determined the exact human threshold a human can sense changes in resistance. Using half of the 20% value, a MATLAB script was created to determine what angles resulted in 10% ± 5% increases in resistance. This resulted in 85 unique states. Because of this, the resistance system shall have no less than 85 states of resistance.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ResistanceSystem.md]: State the constraint and where it came from briefly.

File: Documentation/Signoffs/ResistanceSystem.md [hunk 14]
```diff
@@ -0,0 +1,203 @@
+# Team 4 – Mario Kart & Ride Replay Simulator
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ResistanceSystem.md]: This bears no resemblance to what we have talked about recently. There is no buildable schematic (mechanical or electrical) for any of the systems involved. 

I see a 3d model. But it is very far from being 3rd party buildable.

File: Documentation/Signoffs/ResistanceSystem.md (hunk 8 above)
Comment (branchworm) [reply Documentation/Signoffs/ResistanceSystem.md]: Done; assuming you were referring to the member listing and not the title.

Comment (JesseTNRoberts) [reply Documentation/Signoffs/ResistanceSystem.md]: Best to mark as resolved when something has been addressed.

File: Documentation/Signoffs/ResistanceSystem.md [hunk 15]
```diff
@@ -1,76 +1,138 @@
...
+The function of the resistance system is to control and vary the resistance felt by the user of the Mario Kart Bike, hereafter referred to as the bike. 
 
-**_Constraints:_**
+•	The H-Bridge microcontroller receives a wireless input from the RRS processing subsystem in the form of a numbered value ranging from 0 to 85, which represents the resistance level to be experienced by the user.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ResistanceSystem.md]: This seems like it might not consider the speed the flywheel is turning in choosing the distance.

File: Documentation/Signoffs/ResistanceSystem.md [hunk 16]
```diff
@@ -1,76 +1,138 @@
...
 
-_Maximum Produced Torque:_
+### Constraints:
 
-Based on a study conducted by R C Haut [6], the force required to fracture the patella or femur is approximately 8.5 kN. Applying the gear ratio of the Mario Kart bike and the radius of its wheel, we come to a torque of 4748.94 Nm (torque = gear ratio * wheel radius * force) required to fracture these leg bones. Creating an upper limit of 10% of this value, we get an upper limit torque of 474.89 Nm. Our produced torque utilizing the eddy current braking system shall not exceed this value to protect the user from bodily harm from changes in resistance.
+*Minimum Flywheel Speed:*
 
-**_Buildable schematic:_**
+•	While the user is pedaling the bike at minimum speed, the resistance system must be capable of varying the torque, which will result in the varying of resistance experienced by the rider.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ResistanceSystem.md]: Varying the torque at all? What is the maximum resistance required to be possible at a minimum speed?

File: Documentation/Signoffs/ResistanceSystem.md [hunk 17]
```diff
@@ -1,76 +1,138 @@
...
-![image](https://user-images.githubusercontent.com/114370750/204956463-ca256803-743d-4289-af42-5ac5a970c525.png)
+*Maximum Produced Torque:*
 
-Figure 4. Bracket Dimensions
+•	The potential resistance generated (braking torque) will increase as the bike tire speed increases, resulting in the need to restrict the maximum amount of resistance felt by the user. 
 
+•	Based on a study conducted by R C Haut [4], the force required to fracture the patella or femur is approximately 8.5 kN. 
 
-**_Analysis:_**
+•	Applying the gear ratio of the bike and the radius of its wheel, a torque of 4748.94 N*m would be required to injure a rider.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ResistanceSystem.md]: I don't understand what this is trying to convey.

File: Documentation/Signoffs/ResistanceSystem.md [hunk 18]
```diff
@@ -1,76 +1,138 @@
...
 
+•	The maximum amount of resistance generated at 288 rpm (max bike speed) will be 71.5 N*m.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ResistanceSystem.md]: Based on what? at what distance? These considerations seem good, but they seem to be out of thin air.

File: Documentation/Signoffs/ResistanceSystem.md (hunk 17 above)
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ResistanceSystem.md]: I've not done the math. But this makes no sense.

**Review by JesseTNRoberts — COMMENTED** [review]
There are a great many constraints here. They are somewhat idiosyncratic and hard to follow.

State very clearly the constraints in their entirety. One per bullet. Do not have 5 bullets that may add together to form a constraint.

Is this the existing flywheel or is the flywheel being changed out? 

I didn't see anything about the gear ratio, only a claim that with it thus and such torque is produced. You have to substantiate your claims. 

It seems the resistance max at the minimum speed is not considered. Nor is the min resistance at the max speed. I gave this feedback previously. 

To be clear, the design is greatly improved and I think you are close. But this needs to be cleaned up so that it makes sense to your customer. As it is, it seems plausible that this could work but I'm unconvinced.

File: Documentation/Signoffs/ResistanceSystem.md [hunk 19]
```diff
@@ -0,0 +1,366 @@
...
+Team Members: Ray Durlin, Blake Pickett, Tyler Chittum, Benjamin Reed, and Sage Mooneyham
+
+## Detail Design: Resistance System ##
+
+
+### Function of the Subsystem:
+
+The function of the resistance system is to control and vary the resistance felt by the user of the Mario Kart Bike, hereafter referred to as the bike.  
+
+- The H-Bridge microcontroller receives a wireless input from the RRS processing subsystem in the form of a numbered value ranging from 0 to 85 (this value being calculated using current wheel speed as well as required torque) which represents the distance the magnets are set to. Each magnet distance correlates to a different resistance state.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ResistanceSystem.md]: Then it isn't possible to actually use the 85 states for all valid wheel speeds. Constraint violation.

File: Documentation/Signoffs/ResistanceSystem.md [hunk 20]
```diff
@@ -0,0 +1,366 @@
...
+
+### Function of the Subsystem:
+
+The function of the resistance system is to control and vary the resistance felt by the user of the Mario Kart Bike, hereafter referred to as the bike.  
+
+- The H-Bridge microcontroller receives a wireless input from the RRS processing subsystem in the form of a numbered value ranging from 0 to 85 (this value being calculated using current wheel speed as well as required torque) which represents the distance the magnets are set to. Each magnet distance correlates to a different resistance state.  
+
+- The H-Bridge microcontroller takes an analog input from the linear actuator to determine its current position. 
+
+- An external digital input (via the Bluetooth module) is received to determine the calculated set point which the actuator should try to reach based on the received resistance level.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ResistanceSystem.md]: This is not clear.

File: Documentation/Signoffs/ResistanceSystem.md [hunk 21]
```diff
@@ -0,0 +1,366 @@
...
+
+#### Magnet Distance Range: 
+
+- A distance range of 2-20mm from magnet to flywheel was selected, as these distances matched up well with the torque values that we want the resistance system to apply. 
+
+ 
+
+#### Minimum Flywheel Speed: 
+
+- The minimum speed of the bike tire shall be considered 116.35 rpm. This value is based on the American Council on Exercise's recommendation for the minimum speed required for moderate exercise on a bike [2]. At speeds less than 116.35 rpm, the magnet distance will be set to max, essentially providing the least possible resistance.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ResistanceSystem.md]: This is not clear. Is that the pedal speed?

File: Documentation/Signoffs/ResistanceSystem.md [hunk 22]
```diff
@@ -0,0 +1,366 @@
...
+
+ 
+
+#### Maximum Flywheel Speed: 
+
+- The maximum speed of the bike tire shall be considered 288 rpm, a value that comes from the average pedaling rpm (118 rpm) of an elite level cyclist [3] (Wheel to pedal gear ratio = 2.444, so wheel speed = 2.444*pedal speed). If the user passes this speed, no additional torque will be applied to avoid injury to rider.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ResistanceSystem.md]: This says nothing about the flywheel.

**Review by JesseTNRoberts — COMMENTED** [review]
The constraints are incorrect regarding the speed of the flywheel and do not meet the 85 states constraint.

It is impossible to design something without a clear understanding of the constraints.

File: Documentation/Signoffs/ResistanceSystem.md (hunk 19 above)
Comment (sagemooneyham) [reply Documentation/Signoffs/ResistanceSystem.md]: Why is that? Each speed can have 85 states of resistance due to being able to set the flywheel to magnet distance to 85 different values. Each distance change creates a new state of resistance as the torque changes with magnet to flywheel distance.

Comment (JesseTNRoberts) [reply Documentation/Signoffs/ResistanceSystem.md]: Show that there are 85 valid states at the slowest speed and 85 valid states at the fastest. In the process of doing so, you'll figure out why.

File: Documentation/Signoffs/ResistanceSystem.md [hunk 23]
```diff
@@ -0,0 +1,387 @@
...
+
+ 
+
+#### Maximum Tire Speed: 
+
+- The maximum speed of the bike tire shall be considered 288 rpm, a value that comes from the average pedaling rpm (118 rpm) of an elite level cyclist [3] (Wheel to pedal gear ratio = 2.444, so wheel speed = 2.444*pedal speed). This correlates to 6192 rpm in the flywheel due to the the rotational velocity ratio of 43:2 between the flywheel and the wheel. If the user passes this speed, no additional torque will be applied to avoid injury to rider. 
+
+ 
+
+#### Maximum Torque at Minimum Speed:
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ResistanceSystem.md:44]: This makes sense. My question is, is that sufficient? By that I mean, does that meet your requirements?

File: Documentation/Signoffs/ResistanceSystem.md [hunk 24]
```diff
@@ -0,0 +1,387 @@
...
+•	The drive wheel at the back of the UNISKY bike stand will be screwed into the rear wheel of the bike.
+
+&ensp;&thinsp; &ensp;&thinsp; •	This will ensure maximum transfer of energy when the magnetic breaking occurs to the flywheel. 
+   
+•	The UNISKY bike stand will come with a magnetic brake which will be removed. 
+
+
+*Bracket:*
+
+•	For the magnetic portion of the resistance system, two mounting brackets will be welded to a metal block stand.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/ResistanceSystem.md:116]: You need to use extrusion or tubing. A metal block won't work. Further, it seems you aren't considering the force that will be applied to the actuator (Newton's first law). This will need to be attached to the bike stand or it will be flung away from the bike.

However, I suggest that the mounting system for the actuator be a separate subsystem.
